targets/TARGET_STM/TARGET_STM32G4/analogin_device.c

```diff
@@ -153,11 +163,15 @@ uint16_t adc_read(analogin_t *obj)
 
     // Configure ADC channel
     sConfig.Rank         = ADC_REGULAR_RANK_1;
-    sConfig.SamplingTime = ADC_SAMPLETIME_24CYCLES_5;
+    sConfig.SamplingTime = ADC_SAMPLETIME_247CYCLES_5;
```

Comment:
Really need so much time ?

Reply:
If my math is correct, with the PCLK @ 180MHz and the ADC prescaler set to DIV2, a sample time of 24.5 clock cycles is only ~272 _nanoseconds_... I think this is quite short for most of the slow signals Mbed users will be trying to read with the simple `AnalogIn` API.

My comment from before:

> Also, since the clock speed on this target is so high by default I am seeing small offsets in my ADC readings due to the short accumulation time.
> Setting the ADC prescaler to divide by 4 and setting the sampling time to 640.5 cycles greatly improves the offset but maybe there's a way to get make it so I'm not maxing out the settings?

Perhaps we could increase the ADC prescaler?